### ISSUE TITLE
Multiple outputs/targets

### ISSUE THREAD

**Issue by trunda-o** [jdermody/brightwire#13]
Hello!
Is there a way to train a network that has more than one output? I want to train a regression-network that takes 1000 input features and produces 4 numerical outputs. Training data would in this case have 1004 columns and error would be the sum of MSE errors of the four outputs.

I noticed, that "IDataTable" has a property "TargetColumnIndex" which is "int", so it doesn't seem to support multiple targets. Can I somehow achieve this using "IDataTableVectoriser"?

Thanks.

**Comment by jdermody** [comment]
Hi

Bright Wire's linear regression only supports one single output, however you can use a neural network for regression (be careful about the final activation function btw!).

In this case your training data would consist of a data table with two columns of type vector. The input column is a vector with 1000 elements. The output vector would contain be of size 4.

IDataTableVectoriser assumes one single output column so you should use IDataTableBuilder (via BrightWireProvider.CreateDataTableBuilder) to create a data table directly with the vector columns as described.

You can find an example of creating a data table from vectors in SampleCode/MNIST.cs

Hope this helps.

**Comment by jdermody** [comment]
I've added a multi label sample to brightwire/SampleCode/MultiLabel.cs that might help.

The first example trains a single neural network with six output values. The second builds six networks with each outputting a single value. Both give about the same result.
